lib/dgram.js

```diff
@@ -97,19 +100,24 @@ function newHandle(type, lookup) {
   throw new ERR_SOCKET_BAD_TYPE();
 }
 
-
 function _createSocketHandle(address, port, addressType, fd, flags) {
-  // Opening an existing fd is not supported for UDP handles.
-  assert(typeof fd !== 'number' || fd < 0);
```

Comment:
this used to throw, but now the logic returns an error, so it's a change in API. Given that we export this, it would be a semver-major change.

I'm a bit lost on why we need this `_createSocketHandle`  function at all. Does anyone know? If it's useful, we should remove the `_`. If it's not and it's useful for tests only, we should move it to `internals`.

Reply:
It should be in `internals`. It's hard to say if there is any ecosystem usage though. It's used in core by the `cluster` module.